source/common/upstream/maglev_lb.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "common/upstream/load_balancer_impl.h"
+
+namespace Envoy {
+namespace Upstream {
+
+/**
+ * This is an implementation of Maglev consistent hashing as described in:
+ * https://static.googleusercontent.com/media/research.google.com/en//pubs/archive/44824.pdf
+ * section 3.4. Specifically, the algorithm shown in pseudocode listening 1 is implemented
+ * with a fixed table size of 65537. This is the recommended table size in section 5.3.
+ */
+class MaglevTable {
+public:
+  MaglevTable(const HostVector& hosts);
+
+  HostConstSharedPtr chooseHost(LoadBalancerContext* context);
```

Comment:
For benchmark fairness (at a micro level), this should be virtual (or just inherit LoadBalancer).

Reply:
Sure can fix.